src/main/java/edu/kit/ipd/crowdcontrol/objectservice/notification/SQLEmailNotificationPolicy.java

```diff
@@ -0,0 +1,64 @@
+package edu.kit.ipd.crowdcontrol.objectservice.notification;
+
+import edu.kit.ipd.crowdcontrol.objectservice.database.operations.NotificationOperations;
+import edu.kit.ipd.crowdcontrol.objectservice.mail.MailSender;
+import org.jooq.Record;
+import org.jooq.Result;
+
+import java.time.Instant;
+
+/**
+ * The SQLEmailNotificationPolicy checks given queries in a SQL-Database and sends notifications via email.
+ *
+ * @author Simon Korz
+ * @version 1.0
+ */
+public class SQLEmailNotificationPolicy extends NotificationPolicy<Result<Record>> {
+    private MailSender mailSender;
+    private String receiver;
+    private NotificationOperations operation;
+
+
+    /**
+     * @param mailSender an implementation of the MailSender interfac
+     * @param receiver   email address of the receiver
+     * @param operation  instance of the notification operations
+     */
+    public SQLEmailNotificationPolicy(MailSender mailSender, String receiver, NotificationOperations operation) {
+        this.mailSender = mailSender;
+        this.receiver = receiver;
+        this.operation = operation;
+    }
+
+    @Override
+    protected Result<Record> check(Notification notification) {
+        Result<Record> result = operation.runReadOnlySQL(notification.getQuery());
+        if (result.isNotEmpty()) {
+            return result;
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    protected void send(Notification notification, Result<Record> token) {
+        StringBuilder message = new StringBuilder();
+        message.append(notification.getDescription());
+        message.append("\n");
+        for (Record record : token) {
+            message.append(record.toString());
```

Comment:
result.format() should return a string containing all records in the result with "table look".
But in this case it returns an empty table, because it can't parse it for an unspecified record type